src/ServicePulse.Host/vue/src/components/EventItemShort.vue

```diff
@@ -1,109 +1,25 @@
-    "fa-trash": (eventItem.severity === "error" || eventItem.category === "MessageRedirects") && eventItem.category === "MessageRedirects" && eventItem.event_type === "MessageRedirectRemoved",
-  };
-}
-
-function navigateToEvent(eventLogItem) {
-  switch (eventLogItem.category) {
-    case "Endpoints":
-      router.push("/configuration/endpoint-connection");
-      break;
-    case "HeartbeatMonitoring":
-      window.location = "/a/#/endpoints";
-      break;
-    case "CustomChecks":
-      window.location = "/a/#/custom-checks";
-      break;
-    case "EndpointControl":
-      window.location = "/a/#/endpoints";
-      break;
-    case "MessageFailures":
-      var newlocation = "/failed-messages";
-      if (eventLogItem.related_to && eventLogItem.related_to.length > 0 && eventLogItem.related_to[0].search("message") > 0) {
-        newlocation = "/failed-messages" + eventLogItem.related_to[0];
-      }
-      router.push(newlocation);
-      break;
-    case "Recoverability":
-      router.push("/failed-messages");
-      break;
-    case "MessageRedirects":
-      router.push("/configuration/retry-redirects");
-      break;
-    default:
-  }
-}
+import DataView from "../components/DataView.vue";
+import EventLogItem from "../components/EventLogItem.vue";
 </script>
 
 <template>
-  <div class="row events">
-    <div class="col-12">
-      <h6>Last 10 events</h6>
-
-      <div class="row box box-event-item" v-for="eventLogItem in eventLogItems" :key="eventLogItem.id">
-        <div class="col-12" @click="navigateToEvent(eventLogItem)">
-          <div class="row">
-            <div class="col-1">
-              <span class="fa-stack fa-lg">
-                <i class="fa fa-stack-2x" :class="iconClasses(eventLogItem)"></i>
-                <i v-if="eventLogItem.severity === 'error' || eventLogItem.category === 'MessageRedirects'" class="fa fa-o fa-stack-1x fa-inverse" :class="iconSubClasses(eventLogItem)"></i>
-              </span>
-            </div>
-
-            <div class="col-9">
-              <div class="row box-header">
-                <div class="col-12">
-                  <p class="lead">{{ eventLogItem.description }}</p>
-                </div>
-              </div>
-            </div>
-
-            <div class="col-2">
-              <div>
-                <time-since :dateUtc="eventLogItem.raised_at"></time-since>
-              </div>
-            </div>
-          </div>
-        </div>
-      </div>
-
-      <div class="row text-center">
+  <div class="events">
+    <DataView api-url="eventlogitems" :auto-refresh="5000" :itemsPerPage="10" :show-pagination="false">
+      <template #data="items">
         <div class="col-12">
-          <a v-if="eventCount > 10" class="btn btn-default btn-secondary btn-all-events" href="/a/#/events">View all events</a>
+          <h6>Last 10 events</h6>
+          <EventLogItem v-for="item in items" :eventLogItem="item" :key="item.id" />
```

Comment:
point to note: Vue allows you to use for in or for of, but in raw JavaScript they do different things (in iterates through the keys of the object rather than the items in the enumerable). For this reason I'd favour sticking to for of

Reply:
Applied